src/test/javascript/spec/component/modeling-editor/modeling-editor.component.spec.ts

```diff
@@ -225,4 +226,37 @@ describe('ModelingEditorComponent', () => {
         const spanText = statusHint.query(By.css('span'))?.nativeElement?.textContent;
         expect(spanText).toBe('Saving...');
     });
+
+    it('should handle explanation input change', () => {
+        fixture.detectChanges();
+        const spy = jest.spyOn(fixture.componentInstance.explanationChange, 'emit');
+
+        const newExplanation = 'New Explanation';
+        fixture.componentInstance.onExplanationInput(newExplanation);
+
+        expect(spy).toHaveBeenCalledOnce();
+        expect(spy).toHaveBeenCalledWith(newExplanation);
+        expect(fixture.componentInstance.explanation).toBe(newExplanation);
+    });
+
+    it('should assess model for guided tour for all UML types', () => {
+        const guidedTourService = TestBed.inject(GuidedTourService);
+        const subject = new Subject<string>();
+        jest.spyOn(guidedTourService, 'checkModelingComponent').mockImplementation(() => subject.asObservable());
+        fixture.detectChanges();
+        const updateSpy = jest.spyOn(guidedTourService, 'updateModelingResult');
+        updateSpy.mockReturnThis();
+
+        let currentUmlName = personUML.name;
+        subject.next(currentUmlName);
+        expect(updateSpy).toHaveBeenLastCalledWith(currentUmlName, false);
```

Comment:
I think we should assert call counts here as well :) (and in the two cases below)

Reply:
Added the `toHaveBeenCalledTimes` assertion here and to the cases below as well.